src/nuclear/boltzmann.jl

```diff
@@ -1,16 +1,38 @@
 
 """
-    VelocityBoltzmann(temperature, masses::AbstractVector, dims::Dims{2})
+    VelocityBoltzmann(temperature, masses::AbstractVector, dims::Dims{2};  centre = nothing)
 
-Generate a Boltzmann of velocities for each degree of freedom.
+Generate a Boltzmann distribution of velocities for each degree of freedom.
 
 # Arguments
 
 * `temperature` - Atomic units or Unitful
 * `masses` - Vector of masses for each atom
 * `dims` - (ndofs, natoms). `natoms` must equal `length(masses)`
+* `centre` - Vector of dimension ndofs that provides centre of mass velocity offset
 """
-function VelocityBoltzmann(temperature, masses::AbstractVector, dims::Dims{2})
-    return UnivariateArray([VelocityBoltzmann(temperature, masses[I[2]]) for I in CartesianIndices(dims)])
+function VelocityBoltzmann(temperature, masses::AbstractVector, dims::Dims{2};  centre = zeros(dims))
+    return UnivariateArray([VelocityBoltzmann(temperature, masses[I[2]]; centre = centre[I]) for I in CartesianIndices(dims)])
+end
+VelocityBoltzmann(temperature, mass; centre = 0) = Normal(centre, sqrt(austrip(temperature)/mass))
+
+"""
+    +(dist::VelocityBoltzmann, translation::AbstractArray)
+
+Add a permanent translation to a Velocity distribution, e.g. for a centre of mass translation component. 
+"""
+function Base.:+(dist::UnivariateArray, translation::AbstractArray)
+    if size(dist)!=size(translation)
+        throw(DimensionMismatch("Distribution and translation array must have the same size."))
+    end
+    return UnivariateArray([Normal(dist.sampleable[I].μ+translation[I], dist.sampleable[I].σ) for I in CartesianIndices(dist.sampleable)])
+end
+
+"""
+    +(dist::Normal, translation::Number)
+
+Add a permanent translation to a Velocity distribution. 
+"""
+function Base.:+(dist::Normal, translation::Number)
```

Comment:
It's probably better to define a new function for this operation instead of overloading +. This is what they call "type piracy" where you implement methods for functions that you don't own on types that you don't own.

Reply:
Good point. In that case, I'd just remove that function without offering an alternative, since it's easier to re-initialise a single normal distribution with a new $\mu$ than to do it for an entire Matrix.